### ISSUE TITLE
npm link X requires write access

### ISSUE THREAD

**Issue by postwait** [npm/npm#7198]
If you install (say protobuf) globally as user root (or it is delivered via packaging)

```
npm install -g protobuf
```

Then, in your app you utter:

```
npm link protobuf
```

it fails because it attempts to do a rebuild in the globally installed package.

**Comment by othiym23** [comment]
More context at circonus-labs/reconnoiter@17ecfafc53340f9b2ca76140f79ff8831c770ef1. I think we probably want to look at the whole use case and see if there's a way we can make your workflow less painful / awkward.

**Comment by postwait** [comment]
My desired use case is to be able to run:

```
npm install --link
```

and have it link what it can globally and install the rest.  This is critically important for app installs on production machines without compilers.  Native code modules can be packaged and installed and pure javascript ones can continue to be npm installed in a typical local fashion.

**Comment by smikes** [comment]
Related issue: https://github.com/npm/npm/issues/1573 ; `npm link` must run `npm install` for the module in its global location.  However, subsequent calls to `npm link <project>` also run `npm install` in the global location although this should not be necessary.

**Comment by zkat** [comment]
This was fixed by #8501
